multi.c

```diff
@@ -0,0 +1,77 @@
+#include "multi.h"
+
+#include "brute.h"
+#include "queue.h"
+#include "single.h"
+#include <assert.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+void *
+mt_password_check (void *context)
+{
+  mt_context_t *mt_context = (mt_context_t *)context;
+  task_t task;
+  st_context_t st_context = {
+    .hash = mt_context->config->hash,
+    .data = { .initialized = 0 },
+  };
+
+  while (true)
+    {
+      queue_pop (&mt_context->queue, &task);
```

Comment:
Status check

Reply:
> Status check

But we return `S_FAILURE`, if the queue is not active and the loop stops after this, or is that the idea?